src/python/pants/backend/python/util_rules/pex_environment.py

```diff
@@ -247,40 +281,13 @@ def environment_dict(self, *, python_configured: bool) -> Mapping[str, str]:
         return d
 
 
-@dataclass(frozen=True)
-class SandboxPexEnvironment(CompletePexEnvironment):
-    append_only_caches: FrozenDict[str, str]
-
-    @classmethod
-    def create(cls, pex_environment) -> SandboxPexEnvironment:
-        pex_root = PurePath(".cache") / cls._PEX_ROOT_DIRNAME
-        return cls(
-            _pex_environment=pex_environment,
-            pex_root=pex_root,
-            append_only_caches=FrozenDict({cls._PEX_ROOT_DIRNAME: str(pex_root)}),
-        )
-
-
 class WorkspacePexEnvironment(CompletePexEnvironment):
```

Comment:
We do if you hold being explicit about working directory as a goal, which I was. That said, I could churn a bit more and just eliminate the presence of any rule to produce any CompletePexEnvironment.

Reply:
Fixed.

Reply:
Thanks! I would have also been fine with keeping WorkspacePexEnvironment but also having SandboxPexEnvironment. It just seemed odd to have one and not the other.

Reply:
Yeah - that wouldn't really work since they used to be symmetric in requiring no input parameters (well both required just a PexEnvironment) but now are not. You must provide an additional input parameter to get a SandboxPexEnvironment now - the working directory - and so it cannot be a rule parameter like WorkspacePexEnvironment could be and still can be; so this was really the only route.